### ISSUE TITLE
Set left/right selection boundary dialogs cannot display positions in any of the bar formats

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#4568]
### Steps to reproduce

1. Create an audio track.
2. Set the format in the selection bar to bar:beat or bar:beat:tick.
3. Press [ or ] to open the set left/right selection boundary dialog.

### Expected behavior

The position of the start/end of selection shown in the dialog uses the same format as the format in the selection bar.

### Actual behavior

The dialog uses a time format.
If a user tries to change it to a bar format, then they can't as there are no bar options on the numeric control's context menu.

### Audacity Version

latest beta

### Operating system

Windows

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @DavidBailes 's branch build:  audacity-win-3.3.0-alpha-20230419+b742aa7-x64

This branch build appears to properly fix this issue for both the **[** and **]** shortcuts

![image](https://user-images.githubusercontent.com/50205984/233058586-1b959eff-67fd-4869-9b16-46d09caae42a.png)

**Comment by dozzzzer** [comment]
Tested

**Comment by petersampsonaudacity** [comment]
Testing on W10 wiith latest 3.3.0 Beta

This appears to work ok:
![image](https://user-images.githubusercontent.com/50205984/233433765-76d733fe-5fea-4708-94ac-00c6392797e1.png)
